shared/src/uniswap_pool.rs

```diff
@@ -15,11 +15,29 @@ use crate::current_block::{Block, CurrentBlockStream};
 
 const UNISWAP_PAIR_INIT_CODE: [u8; 32] =
     hex!("96e8ac4277198ff8b6f785478aa9a39f403cb768dd02cbee326c3e7da348845f");
-
 const HONEYSWAP_PAIR_INIT_CODE: [u8; 32] =
     hex!("3f88503e8580ab941773b59034fb4b2a63e86dbc031b3633a925533ad3ed2b93");
 const MAX_BATCH_SIZE: usize = 100;
 
+pub trait AmmResource: Send + Sync + 'static {
```

Comment:
Not sure if Resource is a good name here. All it really does no is give us pair addresses. Maybe we can call it AmmPairProvider?